src/vcpkg/base/json.cpp

```diff
@@ -1502,4 +1502,74 @@ namespace vcpkg::Json
         }
         return sv.to_string();
     }
+
+    bool PackagePatternDeserializer::is_package_pattern(StringView sv)
+    {
+        if (PackageNameDeserializer::is_package_name(sv))
```

Comment:
Agree

Reply:
Unfortunately, it is documented via an error message:

https://github.com/microsoft/vcpkg-tool/blob/6967066fd77b8a4c94e0748ab81a5a967ce6bb4e/src/vcpkg/sourceparagraph.cpp#L455-L459

Reply:
I understand it is in the error message but I don't think this counts as documentation unless there is a practical use for such a thing. (As in, I think the rest of the product tried to interpret the packages array literally rather than splitting by dots)

Reply:
It is gone.